### PR TITLE
Feature: password strength checker

### PR DESCRIPTION
**Pull request by Laurin-W**
I added a password strength indicator to the SignupForm. It uses a default scorer (calculating a password score based on password length and character set).
To deactivate, you can pass `null` or `false` to the component's `scorer` attribute.

It is exported so that it is usable by ActivityPods (e.g. to change passwords).

### CODE REVIEW

File: src/frontend/packages/auth-provider/src/index.js [hunk 1]
```diff
@@ -32,5 +32,10 @@ export { default as useCheckPermissions } from './hooks/useCheckPermissions';
 export { default as usePermissionsWithRefetch } from './hooks/usePermissionsWithRefetch';
 export { default as useSignup } from './hooks/useSignup';
 
+// Password scoring
+export { default as PasswordStrengthIndicator } from './components/LocalLoginPage/PasswordStrengthIndicator';
+export { default as validatePasswordStrength } from './components/LocalLoginPage/validatePasswordStrength';
+export * as passwordScorer from './passwordScorer';
```
Comment (srosset81) [inline src/frontend/packages/auth-provider/src/index.js]: I have an error `passwordScorer` not found when building the ActivityPods frontend. It works if I do an export like the other on this page. Maybe `export * as` is only available on some NodeJS versions ? (I use v16.17.1 currently) In which case, it's maybe better do like the others exports:
```js
export default {
  defaultScorer,
  createPasswordScorer,
  passwordStrength,
  defaultOptions
}
```

File: src/frontend/packages/auth-provider/src/components/LocalLoginPage/SignupForm.js [hunk 2]
```diff
@@ -24,7 +27,23 @@ const useStyles = makeStyles(theme => ({
...
 
-const SignupForm = ({ redirectTo, postSignupRedirect, additionalSignupValues, delayBeforeRedirect }) => {
+/**
+ * @param postSignupRedirect
+ * @param additionalSignupValues
+ * @param delayBeforeRedirect
+ * @param {string} redirectTo
+ * @param {Object} passwordScorer Scorer to evaluate and indicate password strength.
+ *  Set to `null` or `false`, if you don't want password strength checks. Default is
+ *  passwordStrength's `defaultScorer`.
```
Comment (srosset81) [inline src/frontend/packages/auth-provider/src/components/LocalLoginPage/SignupForm.js:37]: It's cool that the `passwordScorer` can be set as a prop, but in reality, the only component we pass props to is `LocalLoginPage`. So please add this `passwordScorer` prop to `LocalLoginPage` and pass it down to `SignupForm` and `NewPasswordForm`.

**Review by srosset81 — COMMENTED** [review]
Here are two comments already. ~~I don't manage to launch the ActivityPods frontend with these changes, because I have a `Error: useNavigate() may be used only in the context of a <Router> component.` on the LocalLoginPage. But it's probably not linked with your work...~~ Solved by running `yarn run bootstrap` on the SemApps frontend directory. See https://marmelab.com/react-admin/Routing.html#troubleshooting

File: src/frontend/packages/auth-provider/src/components/LocalLoginPage/SignupForm.js (hunk 2 above)
Comment (Laurin-W) [reply src/frontend/packages/auth-provider/src/components/LocalLoginPage/SignupForm.js:37]: Oh yea, thanks for pointing out!

**Comment by Laurin-W** [comment]
I made the suggested changes and for me it's working, tested on node 16 and 18.

**Review by srosset81 — APPROVED** [review]
Works great, thank you !

**Pull request merged**